oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/backend/MapReducerJdbc.java

```diff
@@ -79,40 +100,13 @@ protected GridOSHEntity readOshCellRawData(ResultSet oshCellsRawData)
       if (this.typeFilter.isEmpty()) {
         return Stream.empty();
       }
-      ResultSet oshCellsRawData = getOshCellsRawDataFromDb(cellIdRange);
-      if (!oshCellsRawData.next()) {
+      var oshCellIds = getOshTableCellIds(cellIdRange);
+      if (oshCellIds.isEmpty()) {
         return Stream.empty();
       }
-      return StreamSupport.stream(Spliterators.spliteratorUnknownSize(
-          new Iterator<GridOSHEntity>() {
-            @Override
-            public boolean hasNext() {
-              try {
-                return !oshCellsRawData.isClosed();
-              } catch (SQLException e) {
-                throw new RuntimeException(e);
-              }
-            }
-
-            @Override
-            public GridOSHEntity next() {
-              try {
-                if (!hasNext()) {
-                  throw new NoSuchElementException();
-                }
-                GridOSHEntity data = readOshCellRawData(oshCellsRawData);
-                if (!oshCellsRawData.next()) {
-                  oshCellsRawData.close();
-                }
-                return data;
-              } catch (IOException | ClassNotFoundException | SQLException e) {
-                throw new RuntimeException(e);
-              }
-            }
-          }, 0
-      ), false);
+      return oshCellIds.stream().map(this::readOneGridCell);
```

Comment:
can you explain what the rationale was behind this change? Previously this executed a single sql query to fetch all affected cell-id-ranges in a single swoop. Now it would fetch every cell individually in single sql queries, which could come with a (large?) performance overhead.

What is the benefit of doing it in this way?

Reply:
The benefit is to close the resources properly, performance wise this should not introduce a large overhead, because for fetching a single cell the indexes could be fully used.
An other benefit is that cellIds is a full list of ids to process and in theory this stream could be very good parallelized. In theory only because for this to work properly `OSHDBJdbc` should not return a single share connection but from a pool of connection. But this refactoring we should do in an separate PR.
And for `h2` databases, smaller queries are better because `h2` does not have a  cursor and always builds up the full resultset (we can see this by increase file size) before returning.

Reply:
ok, that sound legit.

however, this feels like a quite major change. and here it is presented in a pull request which claims to be only about fixing sonarlint issues, where I would assume that there are no changes to any algorithms or internal business logic. Maybe we can split this change off into a separate PR and take a more [in depth](https://github.com/GIScience/oshdb/pull/419#pullrequestreview-728792474) look there?

Reply:
actually it is fixing this [sonar bugs](https://sonarcloud.io/project/issues?id=org.heigit.ohsome%3Aoshdb&issues=AXjvsOpdmEE7-fvh3PD5&open=AXjvsOpdmEE7-fvh3PD5)  by closing the PreparedStatment properly.
The changes around it just where the consequence of this.
But we could life this out and do it in a separate issue/pr.

Reply:
>  we could leave this out and do it in a separate issue/pr.

:+1:

Reply:
I've created #422 for this.